### PR TITLE
UI layer nk - added new endpoint

### PR DESCRIPTION
**Pull request by wollentimer**

### CODE REVIEW

**Review by CatholicCardinal — APPROVED** [review]

**Review by AnnKirienko — APPROVED** [review]

File: BLL/Services/CityService.cs [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+        private protected IUnitOfWork _db;
+
+        public CityService(IUnitOfWork db)
+        {
+            _db = db;
+        }
+
+        public string GetLocationById(int id)
+        {
+            var city = _db.Cities.Get(id).CityName;
```
Comment (VasiliNesterenko) [inline BLL/Services/CityService.cs:22]: too many calls to the DB...
you can retrieve the city object and then use its properties like id, name and etc
the same issue with the states

File: UI/Controllers/MentorController.cs [hunk 2]
```diff
@@ -17,52 +17,55 @@ public class MentorController : ControllerBase
...
+            _englishLevelService = englishLevelService;
+            _cityService = cityService;
+            _mapper = mapper;
         }
 
-        // GET: api/<MentorController>
-        [HttpGet]
+        [HttpGet("candidates")]
         [Authorize(Roles = "admin, mentor")]
         public async Task<List<CandidateDTO>> GetCandidates()
```
Comment (VasiliNesterenko) [inline UI/Controllers/MentorController.cs:48]: all things related to the candidates should be placed in appropariate locations, like CandidateController
UI will be confused with your endpoints:
api/mentor/getCandidates
api/candidate/getCandidates

**Review by VasiliNesterenko — DISMISSED** [review]
Please find my comments

File: UI/Controllers/MentorController.cs [hunk 3]
```diff
@@ -17,52 +17,55 @@ public class MentorController : ControllerBase
...
-            return "value";
-        }
-
-        // POST api/<MentorController>
-        [HttpPost]
-        public void Post([FromBody] string value)
+        [HttpGet("{id}/form")]
+        [Authorize(Roles = "admin, mentor")]
+        public CandidateForMentorDTO GetForm(int id)
         {
```
Comment (VasiliNesterenko) [inline UI/Controllers/MentorController.cs:60]: why you do not use CandidateDTO?
do you realy need to use string representation of city, specialization and english level?

**Review by Alexy-Def — APPROVED** [review]

**Review by CatholicCardinal — APPROVED** [review]

**Pull request merged**